JeMPI_Apps/JeMPI_API/pom.xml

```diff
@@ -166,7 +166,34 @@
             <artifactId>LibMPI</artifactId>
             <version>1.0-SNAPSHOT</version>
         </dependency>
-
+        <!-- https://mvnrepository.com/artifact/com.softwaremill.akka-http-session/core -->
+        <dependency>
+            <groupId>com.softwaremill.akka-http-session</groupId>
+            <artifactId>core_2.13</artifactId>
+            <version>0.7.0</version>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/org.keycloak/keycloak-adapter-core -->
+        <dependency>
+            <groupId>org.keycloak</groupId>
```

Comment:
Doesn't this tie us to keycloak? Is there not a generic adapter that we can use? Or is this better for some reasons?

Reply:
It's just a library that offers classes we use to interact with keycloak. I made sure that we're not tight to it ;)

Reply:
I think that what Ryan is saying that if we want to change to a different authorisation library (other than Keycloak) that we would need to change a lot of code. 
Currently there is at least one package name and many class names that have the word "keycloak" in them as well as lots of code that has the word "keycloak"
And that's the "tie in" we should probably want to avoid (if possible).
Having "keycloak" in the Maven file is the least of our tie ins.

Another way of possibly doing it would be to create a set of interface classes in JeMPI that reference Oath2 (only).
And then to create a new project that is the Keycloak specific implementation of those interfaces.
The Keycloak specific project could live outside of JeMPI - probably in the Platform family as a general utility project.
(and probably the interfaces could also live outside JeMPI)
That way we could more easily change to a different authorisation library with minimal fuss.

What do you think?

Reply:
Yes, I get the point now thanks. I didn't find any generic adapter that is compatible with our current setup (Akka). All I found is "akka-http-session" which offers a pretty much basic session module and does not offer support for 3rd party OAuth : https://github.com/softwaremill/akka-http-session

In addition keycloak do not offer support for Akka, here are the list of : https://www.keycloak.org/docs/latest/securing_apps/#java-adapters

So I've used the keycloak adapter core package in order to get to use at least some of the helpful classes : https://github.com/keycloak/keycloak/tree/main/adapters/oidc
  
I understand well that this needs to be generic, but there's trade-off to be made here, either we invest time now to make it so or try to make progress on the MVP and refactor in the future as we add other auth methods, knowing  that I already spent a whole week on this.